### PR TITLE
[Connector API][Docs] Document update connector filtering action

### PR DESCRIPTION
**Pull request by jedrazb**
### Changes
- Add documentation for update configuration action

This is the last documentation PR related to Connectors API 🤞

### CODE REVIEW

File: docs/reference/connector/apis/update-connector-filtering-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+`filtering`::
+(Required, array) The filtering configuration for the connector. This configuration determines the set of rules applied for filtering data during syncs.
+
+Each entry in the `filtering` array represents a set of filtering rules for a specific data domain and includes the following attributes:
+
+- `domain` (Required, string) +
+Specifies the data domain to which these filtering rules apply.
+
+- `active` (Required, object) +
```
Comment (jedrazb) [inline docs/reference/connector/apis/update-connector-filtering-api.asciidoc:41]: @leemthompo I used `-` to mark different indentation level vs. `*` - it works as expected, please let me know if there is a different best practice regarding lists with multiple indentation levels. (I remember you suggested changing `-` to `*` in other PR for a different reason I think)

Comment (leemthompo) [reply docs/reference/connector/apis/update-connector-filtering-api.asciidoc:41]: This seems to work fine, handwriting these docs is a nightmare for nested stuff, hopefully will have autogen from spec in the not too distant future

Comment (leemthompo) [reply docs/reference/connector/apis/update-connector-filtering-api.asciidoc:41]: You can always check the [asciidoctor docs](https://docs.asciidoctor.org/asciidoc/latest/lists/unordered/) if in doubt

File: docs/reference/connector/apis/update-connector-filtering-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,186 @@
+[[update-connector-filtering-api]]
+=== Update connector filtering API
+
+preview::[]
+
+++++
+<titleabbrev>Update connector filtering</titleabbrev>
+++++
+
+Updates the `filtering` configuration of a connector.
```
Comment (leemthompo) [inline docs/reference/connector/apis/update-connector-filtering-api.asciidoc:10]: Should we link to [sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html) overview here?

File: docs/reference/connector/apis/update-connector-filtering-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    The value to be used in conjunction with the rule for matching the contents of the document's field.
+    ** `order` (Required, number) +
+    The order in which the rules are applied. The first rule to match has its policy applied.
+    ** `created_at` (Optional, datetime) +
+    The timestamp when the rule was added.
+    ** `updated_at` (Optional, datetime) +
+    The timestamp when the rule was last edited.
+
+  * `advanced_snippet` (Optional, object) +
+  Used for advanced filtering at query time, with the following sub-attributes:
```
Comment (leemthompo) [inline docs/reference/connector/apis/update-connector-filtering-api.asciidoc:64]:
```suggestion
  Used for {enterprise-search-ref}/sync-rules.html#sync-rules-advanced[advanced filtering] at query time, with the following sub-attributes:
```

Link to https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-advanced?

File: docs/reference/connector/apis/update-connector-filtering-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    Indicates the validation state: "edited", "valid", or "invalid".
+    ** `errors` (Optional, object) +
+    Contains details about any validation errors, with sub-attributes:
+      *** `ids` (Required, string) +
+      The ID(s) of any rules deemed invalid.
+      *** `messages` (Required, string) +
+      Messages explaining what is invalid about the rules.
+
+- `draft` (Optional, object) +
+An object identical in structure to the `active` object, but used for drafting and editing filtering rules before they become active.
```
Comment (leemthompo) [inline docs/reference/connector/apis/update-connector-filtering-api.asciidoc:84]: Cool 😎

File: docs/reference/connector/apis/update-connector-filtering-api.asciidoc (hunk 2 above)
Comment (leemthompo) [reply docs/reference/connector/apis/update-connector-filtering-api.asciidoc:10]: Something like "Learn more about filtering in the `{enterprise-search-ref}/sync-rules.html[sync rules] documentation.`

**Review by leemthompo — APPROVED** [review]
LGTM with a few suggestions about links

**Pull request merged**